pages/admin/class/edit/[id].tsx

```diff
@@ -0,0 +1,387 @@
+    const [maxCapacity, setMaxCapacity] = useState("");
+    const [volunteerCapacity, setVolunteerCapacity] = useState("");
+    const [price, setPrice] = useState("");
+
+    const [classDescription, setClassDescription] = useState(
+        Array(sortedLocale.length).join(".").split("."),
+    );
+    //Used to make sure we don't create a new stripe id when editing the class
+    //If the price changes we'll add another price
+    const [stripeId, setStripeId] = useState("");
+
+    const loadPrograms = async () => {
+        const request = {
+            method: "GET",
+            headers: { "Content-Type": "application/json" },
+        };
+        const response = await fetch("/api/program", request).then((res) => res.json());
+
+        setPrograms(response.data);
+    };
+
+    //Get a class by id
+    const getClass = async (id) => {
+        const request = {
+            method: "GET",
+            headers: { "Content-Type": "application/json" },
+        };
+        const response = await fetch(`/api/class/${id}`, request).then((res) => res.json());
+
+        //If the class exists load the values
+        if (!response.error) {
+            const data = response.data;
+
+            setClassName(data.classTranslation[0].name);
+            setAssociatedProgram(data.program?.programTranslation[0]?.name);
+
+            //update class descriptions
+            setClassDescription([
+                ...data.classTranslation.map((translation) => translation.description),
+            ]);
+
+            setStartDate(data.startDate);
+            setEndDate(data.endDate);
+            setImage(data.imageLink);
+            setId(data.id);
+            setIsArchived(data.isArchived);
+            setRepeatOn(data.weekday);
+            setAge(data.borderAge);
+            setAboveUnder(data.isAgeMinimal ? "Under" : "Above");
+            setMaxCapacity(data.spaceTotal);
+            setVolunteerCapacity(data.volunteerSpaceTotal);
+            setStripeId(data.stripePriceId);
+        }
+    };
+
+    useEffect(() => {
+        loadPrograms();
+        getClass(router.query.id);
+    }, [router.query]);
+
+    async function save() {
+        const classData = {
+            programId: programs.find(
+                (program) => program.programTranslation[0].name === associatedProgram,
+            ).id,
+            imageLink: image,
+            isAgeMinimal: aboveUnder === "Under",
+            spaceTotal: parseInt(maxCapacity),
+            volunteerSpaceTotal: parseInt(volunteerCapacity),
+            stripePriceId: stripeId,
+            borderAge: parseInt(age),
+            weekday: repeatOn,
+            name: className,
+            startDate: new Date(startDate),
+            endDate: new Date(endDate),
+            price,
+            id,
+            //location,
+            //locationDescription,
+            //teacherName,
+
+            //filling in
+            startTimeMinutes: 0,
+            durationMinutes: 0,
```

Comment:
Attention ^

Reply:
@bekahma @vedantdsn looks like in our class creation forms, we are still missing the start time and the duration of the class.

Maybe something like this?

![image](https://user-images.githubusercontent.com/44826218/144611410-9cf2e1af-637d-4b12-a41e-3700db11365c.png)

Reply:
I can just add these in using my best design judgement